base/special/exp.jl

```diff
@@ -1,65 +1,198 @@
-# Based on FDLIBM http://www.netlib.org/fdlibm/e_exp.c
-# which is made available under the following licence
-
-## Copyright (C) 2004 by Sun Microsystems, Inc. All rights reserved. Permission
-## to use, copy, modify, and distribute this software is freely granted,
-## provided that this notice is preserved.
-
-# Method
-# 1. Argument reduction: Reduce x to an r so that |r| <= 0.5*ln(2). Given x,
-#    find r and integer k such that
-#       x = k*ln(2) + r,  |r| <= 0.5*ln(2).
-#    Here r is represented as r = hi - lo for better accuracy.
-#
-# 2. Approximate exp(r) by a special rational function on [0, 0.5*ln(2)]:
-#       R(r^2) = r*(exp(r)+1)/(exp(r)-1) = 2 + r*r/6 - r^4/360 + ...
-#
-#    A special Remez algorithm on [0, 0.5*ln(2)] is used to generate a
-#    polynomial to approximate R.
-#
-#    The computation of exp(r) thus becomes
-#                       2*r
-#       exp(r) = 1 + ----------
-#                     R(r) - r
-#                          r*c(r)
-#              = 1 + r + ----------- (for better accuracy)
-#                         2 - c(r)
-#    where
-#       c(r) = r - (P1*r^2  + P2*r^4  + ... + P5*r^10 + ...).
-#
-# 3. Scale back: exp(x) = 2^k * exp(r)
+# magic rounding constant: 1.5*2^52 Adding, then subtracting it from a float rounds it to an Int.
```

Comment:
Interesting, in the specific it doesn't actually matter. The only reason for 1.5 is it maintains the maximum range of accuracy for positive/negative constants. Since in this case the numbers are all small anyway, anything between 1 and 1.9 *2^52 would work fine. That said, I will fix.

Reply:
OK, that's what I was guessing and am glad you can confirm. Thanks.